Saule/StringExtensions.cs

```diff
@@ -23,7 +23,7 @@ public static string EnsureStartsWith(this string source, string start)
 
         public static string TrimJoin(this char separator, params string[] parts)
         {
-            return string.Join(separator.ToString(), parts.Select(p => p.Trim(separator)));
+            return string.Join(separator.ToString(), parts.Select(p => p?.Trim(separator)));
```

Comment:
Well, all tests passes without this change so I'll revert it and if there is a reason we will find it eventually. :)